app/web_conn.go

```diff
@@ -287,6 +287,14 @@ func (webCon *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 		return false
 	}
 
+	if msg.Broadcast.OmitAdmins && webCon.App.RolesGrantPermission(webCon.GetSession().GetUserRoles(), model.PERMISSION_MANAGE_SYSTEM.Id) {
+		return false
+	}
+
+	if msg.Broadcast.OmitNonAdmins && !webCon.App.RolesGrantPermission(webCon.GetSession().GetUserRoles(), model.PERMISSION_MANAGE_SYSTEM.Id) {
```

Comment:
These should, in theory, be expensive checks as, now and in the future, we should already have the user's roles, the roles themselves and permissions either locally in a memory cache or hard-coded.

Reply:
Can we at least not run the same call out to the permissions system twice in a row here.

Reply:
I can update it but it's only ever going to run twice if both `OmitAdmins` and `OmitNonAdmins` are set which /shouldn't/ ever happen. I can also pull it out completely and just store it on the on the webconn struct

Reply:
Awesome, thanks George. I'll update the variable names to something more appropriate

Reply:
Ya, thanks George!